src/main/java/de/bonndan/nivio/input/ItemRelationProcessor.java

```diff
@@ -66,29 +70,39 @@ public ProcessingChangelog process(@NonNull final LandscapeDescription input, @N
                     .filter(relation -> !processed.contains(relation))
                     .filter(relation -> origin.equals(relation.getSource()))
                     .forEach(relation -> {
-                        processLog.info(String.format(origin + ": Removing relation between %s and %s", relation.getSource(), relation.getTarget()));
-                        Item currentSource = landscape.getItems().pick(
-                                relation.getSource().getFullyQualifiedIdentifier().getItem(),
-                                relation.getSource().getFullyQualifiedIdentifier().getGroup()
-                        );
-                        if (!currentSource.removeRelation(relation)) {
-                            processLog.warn(String.format("Could not remove relation %s from source %s", relation, relation.getSource()));
-                        }
-
-                        Item currentTarget = landscape.getItems().pick(
-                                relation.getTarget().getFullyQualifiedIdentifier().getItem(),
-                                relation.getTarget().getFullyQualifiedIdentifier().getGroup()
-                        );
-                        if (!currentTarget.removeRelation(relation)) {
-                            processLog.warn(String.format("Could not remove relation %s from target %s", relation, relation.getSource()));
-                        }
+                        processLog.info(String.format("%s: Removing relation between %s and %s", origin, relation.getSource(), relation.getTarget()));
+                        removeRelationFromItem(landscape, relation, relation.getSource());
+                        removeRelationFromItem(landscape, relation, relation.getTarget());
                         changelog.addEntry(relation, ProcessingChangelog.ChangeType.DELETED, null);
                     });
         });
 
         return changelog;
     }
 
+    /**
+     * Gracefully finds the relation end item in the landscape and tries to remove the relation.
+     *
+     * @param landscape the current landscape
+     * @param relation the relation to remove
+     * @param relationEnd the relation source or target
+     */
+    private void removeRelationFromItem(Landscape landscape, Relation relation, Item relationEnd) {
+        var fqi = relationEnd.getFullyQualifiedIdentifier();
+        landscape.getItems().find(fqi.getItem(), fqi.getGroup())
+                .ifPresentOrElse(item -> {
+                    if (!item.removeRelation(relation)) {
```

Comment:
Um die Lesbarkeit zu verbessern würde ich item.removeRelation(relation) in eine sprechende Variable auslagern und die dann in der If-Bedingung verwenden

Reply:
Item already has a field `identifier` which is part of the fully qualified identifier (but not unique). So that might be misleading.

Reply:
This is not my point here ;-) Here you call a method within an If clause which executes business logic and not only does a check. In my opinion this is bad practice and not really readable. You could use a variable like `isRelationRemoved` for example and use it in the If clause.

Reply:
Sorry, above I have answered the question below 🤦